pylint/checkers/variables.py

```diff
@@ -1826,6 +1826,10 @@ def _store_type_annotation_node(self, type_annotation):
             self._type_annotation_names.append(type_annotation.name)
             return
 
+        if isinstance(type_annotation, astroid.Attribute):
+            self._store_type_annotation_node(type_annotation.expr)
```

Comment:
I could change this to a more restrictive form

```
if isinstance(type_annotation, astroid.Attribute):
    if not isinstance(type_anntoation.expr, astroid.Name):
         return
    self._type_annotation_names.append(type_annotation.expr.name)
```

Reply:
Hello, could you maybe add a use case so it's easier to understand what this would prevent ?

Reply:
The current check covers all of

```python
# type: foo.Bar
# type: foo.bar.Boo
# type: Foo[T].Bar
# type: Foo[S, T].Bar
```

The restricted check only allows `# type: foo.Bar`. I personally prefer the current check to the restricted one, but I don't feel strongly.

Reply:
Thank you for the explanation :) I also prefer handling all the use cases. Would it be possible to add those 4 examples in the functional tests in `tests/functional` ?

Reply:
I've added them to `variables_test.py`. Would you prefer me also add them to `tests/functional`?

Reply:
I like functional because there's no tests code around it, only the code and the expected output, but as long as there is a test it's ok :)